src/plugins/discourse/config.js

```diff
@@ -1,29 +1,39 @@
 // @flow
 
-import * as Combo from "../../util/combo";
+import * as C from "../../util/combo";
 import {optionsShapeParser, type MirrorOptions} from "./mirror";
+import {
+  type WeightsConfig,
+  weightsConfigParser,
+  type SerializedWeightsConfig,
+} from "./weights";
+
+export type SerializedDiscourseConfig = {|
+  +serverUrl: string,
+  +mirrorOptions?: $Shape<MirrorOptions>,
+  +weights: SerializedWeightsConfig,
+|};
 
 export type DiscourseConfig = {|
   +serverUrl: string,
   +mirrorOptions?: $Shape<MirrorOptions>,
+  +weights?: WeightsConfig,
```

Comment:
Weights should be mandatory in the DiscourseConfig (its optional in the SerializedConfig, fill in with defaults when not present)